yext/answers-headless-react
# Support "Near Me" queries

Add support for 'Near me' queries in search bar

- added `search-operations` file that includes util functions for executing search, getting search intents, and user location
- update LocationBias component to retrieve user location and execute search using SearchHandler.
- updated `VerticalSearchPage` and `UniversalSearchPage` fetch searchIntents from autocomplete search, in the case of a default initial search with near me intent, get and update user's location if there is a near me intent, and execute query.
- updated searchBar to rely on autocomplete response from latest request to retrieve the search intent, and update user's location before executing a search

Note: there's a noticeable wait time when navigator.geolocation is used. and there isn't any indication that the results maybe  in loading state. Should consider exposing a setLoadingStatus in answersHeadless, so it can be use when executeSearchWithUserLocation is triggered. 

J=SLAP-1706
TEST=manual

- set location to washington DC in when vertical page is loaded, click update my location from `LocationBias` component, and see that the label is updated to my device's location.
- set default search query to 'engineers near me' and see that user location is set based on location near my device.
- ran searches between non-nearMe and nearMe queries, see that setUserLocation is only on queries with nearMe intents
